### ISSUE TITLE
reranker训练的logits和推理没有对齐

### ISSUE THREAD

**Issue by Gaodzlearn** [modelscope/ms-swift#8386]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new question or discussion topic. / 我已经搜索过现有的 issues，确认这是一个新的问题与讨论。

### Question Description / 问题描述

qwen3_reranker训练时，代码logits使用的是 yes_token_logit - no_token_logit 的指标，[推理代码](https://github.com/modelscope/ms-swift/blob/a807cb910bf7020f8336e0042b17133d544c52db/swift/utils/torch_utils.py#L294)如下
```
def get_generative_reranker_logits(lm_head_weight, tokenizer, hidden_states):
    positive_token = os.environ.get('GENERATIVE_RERANKER_POSITIVE_TOKEN', 'yes')
    negative_token = os.environ.get('GENERATIVE_RERANKER_NEGATIVE_TOKEN', 'no')
    positive_token_id = tokenizer.convert_tokens_to_ids(positive_token)
    negative_token_id = tokenizer.convert_tokens_to_ids(negative_token)
    weight = lm_head_weight[[positive_token_id, negative_token_id]]
    logits = F.linear(hidden_states, weight)
    return logits[..., 0:1] - logits[..., 1:2]
```
而官方提供的[推理代码](https://huggingface.co/Qwen/Qwen3-Reranker-0.6B)使用的是 softmax(yes_token, no_token)中的yes_token数值：
```
def compute_logits(inputs, **kwargs):
    batch_scores = model(**inputs).logits[:, -1, :]
    true_vector = batch_scores[:, token_true_id]
    false_vector = batch_scores[:, token_false_id]
    batch_scores = torch.stack([false_vector, true_vector], dim=1)
    batch_scores = torch.nn.functional.log_softmax(batch_scores, dim=1)
    scores = batch_scores[:, 1].exp().tolist()
    return scores
```
我在使用listwise_reranker_loss训练时，发现loss降得很低，但abs(yes_token)和abs(no_token)都比较高，虽然他们的差值较大（这符合训练损失的要求），但softmax后差距很小，导致输出的score偏低。
请问区分训练损失和推理logits的计算方法，这样做的考量是什么呢？

**Comment by Jintao-Huang** [comment]
listwise的loss嘛

你试试pointwise的

**Comment by Tord-Zhang** [comment]
@Jintao-Huang 请问训point-wise reranker的时候, 每个sample中都必须有positive_message和negative_message吗? 某个sample
只有positive_message或者negative_message可以吗
